src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -348,6 +348,12 @@ void Assembler::emit_arith_operand(int op1, Register rm, Address adr, int32_t im
   }
 }
 
+void Assembler::emit_arith_operand_imm32(int op1, Register rm, Address adr, int32_t imm32) {
```

Comment:
Why is this needed?

Reply:
@TobiHartmann Thanks for taking a look, method entry barrier needs a 4-byte immediate so it can be patched. This method is used in `Assembler::cmpl_imm32`

Reply:
Okay, so code before emitted a 8-bit immediate. Why didn't that cause any issues when patching?

Reply:
`cmpl(Address, int)` unconditionally emits a 32-bit immediate before, I make it emit 8-bit immediate where possible and create `cmpl_imm32` to strictly emit a 32-bit immediate, similar to how we have `subl_imm32` and `subq_imm32`.

Reply:
Ah! Now I got it. Thanks for the explanation.